client-integration/__tests__/router.spec.js

```diff
@@ -0,0 +1,36 @@
+import getBrowser from "../helpers/getBrowser";
+import { METRICS_URL } from "../config/config";
+
+const tabIds = ["origin", "fork-time", "commissioned-length"];
+
+describe("The Tabs", () => {
+    let page;
+
+    beforeAll(async done => {
+        page = await getBrowser().newPage();
+        await page.goto(METRICS_URL);
+        done();
+    });
+
+    afterAll(async () => {
+        await page.close();
+    });
+
+    it("should load without error", async () => {
+        await page.waitForSelector(".tabs__link");
```

Comment:
I prefer keeping the css selectors separate from the test logic themselves as they are related to the page structure rather than the logic being tested

Reply:
I agree, I'd probably wrap `page` in a closure and just have some methods on that like `tabsLinks`. What does story packages do? Ideally I'd wrap puppeteer's interface up completely in to something more domain specific. It's not made for testing out of the box and it a little too general as it is.

Reply:
Yes, that sounds good. Story packages is facia-tools little sister, it's only helpful in terms of providing inspiration as it's client side tooling is a bit old but here are some [tests](https://github.com/guardian/story-packages/blob/master/public/test/spec/story-packages.spec.js)

Reply:
I've updated this slightly - see what you think!